### DISCUSSION TITLE
How to  inject SavedStateHolder into ViewModel

### DISCUSSION THREAD

**Discussion by corneloaie** [Tlaster/PreCompose#274] [Q&A]
Hello, when I'm using the ViewModel with Koin, I also want the SavedStateHolder which I don't understand exactly how to create instance for it
```
val appViewModel = koinViewModel(vmClass = AppViewModel::class)
val appVm = viewModel(modelClass = AppViewModel::class) {
AppViewModel(AppState(), it)
}
```

How can I achieve with koinViewModel the behaviour viewModel method? Or what should I add in my modules in order to get() inside the AppViewModel?
Also, in docs it's stated "when using PreCompose ViewModel with Koin, you do have to manage ViewModel's lifecycle", then what can I do to achieve the lifecycle managing with Koin vm that is obtained from the classical creation?

**Comment by Tlaster** [comment]
You can pass SavedStateHolder as parameters, there's a `parameters` parameter for koinViewModel.
And the doc said this is when you're not using precompose-koin, when you're using precompose-koin, the lifecycle is handled for you.
